internal/pkg/unit/match_request.go

```diff
@@ -32,7 +32,9 @@ func (sm *ExtendedStringMatch) Match(s string) (bool, error) {
 	case sm.GetPrefix() != "":
 		return strings.HasPrefix(s, sm.GetPrefix()), nil
 	case sm.GetRegex() != "":
-		r, err := regexp.Compile(sm.GetRegex())
+		// The rule will not match if only a subsequence of the string matches the regex.
+		// https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-field-config-route-v3-routematch-safe-regex
+		r, err := regexp.Compile("^" + sm.GetRegex() + "$")
```

Comment:
Should we add a check to avoid adding the start/end markers twice? (eg. if `sm.GetRegex()` already returns a regex surrounded by `^` and `$` ?)

Reply:
I thought about it but it shouldn't matter, right?

Reply:
Indeed, it does not make a difference in regex evaluation.